src/compiler/binder.ts

```diff
@@ -1668,18 +1671,22 @@ namespace ts {
         }
 
         function bindOptionalChainRest(node: OptionalChain) {
-            bind(node.questionDotToken);
             switch (node.kind) {
                 case SyntaxKind.PropertyAccessExpression:
+                    bind(node.questionDotToken);
                     bind(node.name);
                     break;
                 case SyntaxKind.ElementAccessExpression:
+                    bind(node.questionDotToken);
                     bind(node.argumentExpression);
                     break;
                 case SyntaxKind.CallExpression:
+                    bind(node.questionDotToken);
                     bindEach(node.typeArguments);
                     bindEach(node.arguments);
                     break;
+                case SyntaxKind.NonNullExpression:
+                    break;

```

Comment:
There's nothing left to bind for a `NonNullExpression`, since the expression has already been bound. `bindOptionalChainRest` binds whatever's left after the `expression` has been bound.

Reply:
What I meant is why leave the case in?  (Maybe to make it explicit?)